lib/facebook_ads/ad_objects/server_side/custom_data.rb

```diff
@@ -83,54 +87,46 @@ class CustomData
       # @param [Integer] num_items
       # @param [String] status
       # @param [String] search_string
-      def initialize(value: nil,
-                     currency: nil,
-                     content_name: nil,
-                     content_category: nil,
-                     content_ids: nil,
-                     contents: nil,
-                     content_type: nil,
-                     order_id: nil,
-                     predicted_ltv: nil,
-                     num_items: nil,
-                     status: nil,
-                     search_string: nil)
+      def initialize(**data)
 
-        unless value.nil?
-          self.value = value
+        unless data[:value].nil?
+          self.value = data.delete(:value)
         end
-        unless currency.nil?
-          self.currency = currency
+        unless data[:currency].nil?
+          self.currency = data.delete(:currency)
         end
-        unless content_name.nil?
-          self.content_name = content_name
+        unless data[:content_name].nil?
+          self.content_name = data.delete(:content_name)
         end
-        unless content_category.nil?
-          self.content_category = content_category
+        unless data[:content_category].nil?
+          self.content_category = data.delete(:content_category)
         end
-        unless content_ids.nil?
-          self.content_ids = content_ids
+        unless data[:content_ids].nil?
+          self.content_ids = data.delete(:content_ids)
         end
-        unless contents.nil?
-          self.contents = contents
+        unless data[:contents].nil?
+          self.contents = data.delete(:contents)
         end
-        unless content_type.nil?
-          self.content_type = content_type
+        unless data[:content_type].nil?
+          self.content_type = data.delete(:content_type)
         end
-        unless order_id.nil?
-          self.order_id = order_id
+        unless data[:order_id].nil?
+          self.order_id = data.delete(:order_id)
         end
-        unless predicted_ltv.nil?
-          self.predicted_ltv = predicted_ltv
+        unless data[:predicted_ltv].nil?
+          self.predicted_ltv = data.delete(:predicted_ltv)
         end
-        unless num_items.nil?
-          self.num_items = num_items
+        unless data[:num_items].nil?
+          self.num_items = data.delete(:num_items)
         end
-        unless status.nil?
-          self.status = status
+        unless data[:status].nil?
+          self.status = data.delete(:status)
         end
-        unless search_string.nil?
-          self.search_string = search_string
+        unless data[:search_string].nil?
+          self.search_string = data.delete(:search_string)
+        end
+        unless data.keys.empty?
+          self.custom_properties = data
```

Comment:
so this assumes the above order needs to be followed.

might be safer to add `custom_properties` as a hash on the signature?

Reply:
Decide to switch back